plugins/base_data_quality_operator.py

```
@@ -64,28 +54,28 @@ def conn_type(self, conn):
         if conn not in conn_types:
             raise ValueError(f"""Connection type of "{conn}" not currently supported""")
         self._conn_type = conn
-
-    def _get_hook(self, conn_type, conn_id):
-        if conn_type == "postgres":
-            return PostgresHook(postgres_conn_id=conn_id)
-        if conn_type == "mysql":
-            return MySqlHook(mysql_conn_id=conn_id)
-        if conn_type == "hive":
-            return HiveServer2Hook(hiveserver2_conn_id=conn_id)
-
-    def get_result(self, conn_type, conn_id, sql):
-        hook = self._get_hook(conn_type, conn_id)
-        result = hook.get_records(sql)
-        if len(result) > 1:
-            logging.info("Result: %s contains more than 1 entry", str(result))
-            raise ValueError("Result from sql query contains more than 1 entry")
-        if len(result) < 1:
-            raise ValueError("No result returned from sql query")
-        if len(result[0]) != 1:
-            logging.info("Result: %s does not contain exactly 1 column", str(result[0]))
-            raise ValueError("Result from sql query does not contain exactly 1 column")
-        return result[0][0]
-
+    
     def push(self, info_dict):
         """Send data check info and metadata to an external database."""
         raise NotImplementedError()
+
+def _get_hook(conn_type, conn_id):
+    if conn_type == "postgres":
+        return PostgresHook(postgres_conn_id=conn_id)
+    if conn_type == "mysql":
+        return MySqlHook(mysql_conn_id=conn_id)
+    if conn_type == "hive":
+        return HiveServer2Hook(hiveserver2_conn_id=conn_id)
```

Can you throw the exception here  raise ValueError(f"""Connection type of "{conn}" not currently supported""")  . Then you don't have to do the checks in the inherited classes.

---

I want to include the exception raise in `DataQualityThresholdSQLOperator` so that the error could be raised at object construction instead of mid execution. But i could also add the exception raise to _get_hook as well